frame/democracy/src/lib.rs

```diff
@@ -0,0 +1,2084 @@
+		///
+		/// Also used to compute weight, an overly big value can
+		/// lead to extrinsic with very big weight: see `delegate` for instance.
+		#[pallet::constant]
+		type MaxVotes: Get<u32>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+
+		/// The maximum number of public proposals that can exist at any time.
+		#[pallet::constant]
+		type MaxProposals: Get<u32>;
+	}
+
+	// TODO: Refactor public proposal queue into its own pallet.
+	// https://github.com/paritytech/substrate/issues/5322
+	/// The number of (public) proposals that have been made so far.
+	#[pallet::storage]
+	#[pallet::getter(fn public_prop_count)]
+	pub type PublicPropCount<T> = StorageValue<_, PropIndex, ValueQuery>;
+
+	/// The public proposals. Unsorted. The second item is the proposal's hash.
+	#[pallet::storage]
+	#[pallet::getter(fn public_props)]
+	pub type PublicProps<T: Config> = StorageValue<
+		_,
+		Vec<(PropIndex, ProposalId<T::Hash, T::AssetId>, T::AccountId)>,
+		ValueQuery,
+	>;
+
+	/// Those who have locked a deposit.
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn deposit_of)]
+	pub type DepositOf<T: Config> =
+		StorageMap<_, Twox64Concat, PropIndex, (Vec<T::AccountId>, BalanceOf<T>)>;
+
+	/// Map of hashes to the proposal preimage, along with who registered it and their deposit.
+	/// The block number is the block at which it was deposited.
+	// TODO: Refactor Preimages into its own pallet.
+	// https://github.com/paritytech/substrate/issues/5322
+	#[pallet::storage]
+	pub type Preimages<T: Config> = StorageMap<
+		_,
+		Identity,
+		ProposalId<T::Hash, T::AssetId>,
+		PreimageStatus<T::AccountId, BalanceOf<T>, T::BlockNumber>,
+	>;
+
+	/// The next free referendum index, aka the number of referenda started so far.
+	#[pallet::storage]
+	#[pallet::getter(fn referendum_count)]
+	pub type ReferendumCount<T> = StorageValue<_, ReferendumIndex, ValueQuery>;
+
+	/// The lowest referendum index representing an unbaked referendum. Equal to
+	/// `ReferendumCount` if there isn't a unbaked referendum.
+	#[pallet::storage]
+	#[pallet::getter(fn lowest_unbaked)]
+	pub type LowestUnbaked<T> = StorageValue<_, ReferendumIndex, ValueQuery>;
+
+	/// Information concerning any given referendum.
+	///
+	/// TWOX-NOTE: SAFE as indexes are not under an attacker’s control.
+	#[pallet::storage]
+	#[pallet::getter(fn referendum_info)]
+	pub type ReferendumInfoOf<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ReferendumIndex,
+		ReferendumInfo<T::BlockNumber, T::Hash, BalanceOf<T>, T::AssetId>,
+	>;
+
+	/// All votes for a particular voter. We store the balance for the number of votes that we
+	/// have recorded. The second item is the total amount of delegations, that will be added.
+	///
+	/// TWOX-NOTE: SAFE as `AccountId`s are crypto hashes anyway.
+	#[pallet::storage]
+	pub type VotingOf<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		(T::AccountId, T::AssetId),
```

Comment:
please update docs comments for using 2 keys.

Reply:
Done